ComponentKit/Utilities/CKArrayControllerChangeset.h

```diff
@@ -207,23 +207,36 @@ namespace CK {
     namespace Output {
 
       struct Change {
+        /** soon to be removed, use sourceIndexPath/destinationIndexPath instead */
         CKArrayControllerIndexPath indexPath;
+        
+        /** Valid for updates and removals. */
+        CKArrayControllerIndexPath sourceIndexPath;
+        /** Valid for insertions. */
+        CKArrayControllerIndexPath destinationIndexPath;
+
         id<NSObject> before;
         id<NSObject> after;
 
-        Change(const CKArrayControllerIndexPath &iP, id<NSObject> b, id<NSObject> a) : indexPath(iP), before(b), after(a) {};
+        Change(const CKArrayControllerIndexPath &sIP, const CKArrayControllerIndexPath &dIP, id<NSObject> b, id<NSObject> a) : indexPath((sIP.section != NSNotFound)?sIP:dIP), sourceIndexPath(sIP), destinationIndexPath(dIP), before(b), after(a) {};
 
         bool operator==(const Change &other) const {
-          return indexPath == other.indexPath && CKObjectIsEqual(before, other.before) && CKObjectIsEqual(after, other.after);
+          return sourceIndexPath == other.sourceIndexPath && destinationIndexPath == other.destinationIndexPath && CKObjectIsEqual(before, other.before) && CKObjectIsEqual(after, other.after);
         }
 
         bool operator<(const Change &other) const {
```

Comment:
I'm not opposed to removing this, but I'm not sure why it made sense before but it doesn't now. Also to really compare two Changes you have to do this exact full comparison, so why not keep it?